### PR TITLE
Update the description for the complex-numbers exercise

### PR DESCRIPTION
**Pull request by jagdish-15**
### Pull Request Description

This pull request updates the description file for the **Complex Number** exercise, as discussed in the [forum thread](https://forum.exercism.org/t/guidance-needed-for-adding-new-test-cases-for-complex-number-exercise/13649/13). The changes reflect the updated description with no objections raised during the discussion.

### CODE REVIEW

**Comment by Cool-Katt** [comment]
@jagdish-15 Looks like you might have forgotten to run the linter on this.

**Comment by jagdish-15** [comment]
@Cool-Katt, apologies for the initial oversight!

I’ve updated the file so it now passes all tests. I’m a bit uncertain about the ordered list formatting—I've prefixed each item with “1.” as required by markdownlint, but if there’s a better approach, I'd appreciate any guidance. Thanks!

File: exercises/complex-numbers/description.md
Comment (SleeplessByte) [inline exercises/complex-numbers/description.md:19]: Instead of making this a list, try:

```markdown
### Conjugate

[...]

```

For example below this line:


-------------------------------

## Key Properties

### Conjugate

The conjugate of the complex number `z = a + b * i` is given by:

```math
z̅ = a - b * i
```

### Absolute Value

The absolute value (or modulus) of `z` is defined as:

```math
|z| = sqrt(a^2 + b^2)
```

**Comment by tasxatzial** [comment]
@jagdish-15 Please replace the `1.` with just `-`. This is an unordered list.

**Review by SleeplessByte — COMMENTED** [review]
For those who want to see the rich text [in the latest form](https://github.com/jagdish-15/problem-specifications/blob/chnage-description-complex/exercises/complex-numbers/description.md) when reviewing. And [here is the split raw view without whitespace changes](https://github.com/exercism/problem-specifications/pull/2496/files?diff=split&w=1#diff-e6831cd784873c50d20d539bd9fc87ee4efa9d03439180ffc54bbe2b5fbcd10c)

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:19]: @jagdish-15 Go with SleeplessByte's suggestion, disregard mine.

**Review by SleeplessByte — APPROVED** [review]
Tentatively approved if you fix the linter.

I think this is better than what it was, so yah.

**Comment by jagdish-15** [comment]
Thank you so much for the feedback, @SleeplessByte! I'll definitely address the linter issues. If you notice any other areas for improvement or have any suggestions for refining this version further, I’d love to hear them!

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [inline exercises/complex-numbers/description.md:9]: Doesn't let me apply a suggestion, so I'm leaving it as a comment:

Consider replacing the selected text with:

A **complex number** is expressed in the form `z = a + b * i`, where:

**Comment by jagdish-15** [comment]
I’d also love to know if this change will automatically update in instructions.md across all tracks, or if it’ll require a manual update.

**Comment by BNAndras** [comment]
> I’d also love to know if this change will automatically update in instructions.md across all tracks, or if it’ll require a manual update.

Each track will need to sync it manually. Some tracks do have tooling to alert the maintainer when there are unsynced changes in the problem specifications repo, but the actual syncing is still done by a human.

**Comment by tasxatzial** [comment]
It is possible to trigger a PR that updates the description across all tracks with a script. However, it's more likely that track maintainers will perform a batch sync once a month, for example, to pull in all changes from this repo.

**Review by Cool-Katt — APPROVED** [review]

**Comment by jagdish-15** [comment]
Thank you so much, everyone!

Before wrapping up, I’d love to know if there’s anything I should keep in mind for next time, or if you’d like any additional changes to the current file.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [inline exercises/complex-numbers/description.md:9]:
```suggestion
- `i` is the **imaginary unit** satisfying `i^2 = -1`.
```

**Review by tasxatzial — COMMENTED** [review]
One small change for consistency

**Comment by jagdish-15** [comment]
Thank you so much, @tasxatzial! I really appreciate all your help. If there’s anything else that might need fixing, I’d love to hear your suggestions.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [inline exercises/complex-numbers/description.md:95]:
```suggestion
- **exponentiation** of the base _e_ (the natural logarithm) to a complex number
```

**Review by tasxatzial — COMMENTED** [review]
Hopefully the final suggestion. I just realized that the "exponent of a complex" number isn't even proper math. Let's fix that.

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:95]: Makes sense!!

**Comment by jagdish-15** [comment]
Done!

**Review by tasxatzial — APPROVED** [review]

**Comment by SleeplessByte** [comment]
Now we just need @exercism/maintainers-admin to sign-off. Great work @jagdish-15.

**Review by tasxatzial — COMMENTED** [review]
@jagdish-15 Final suggestion, this is also for consistency with the previous part that introduces exponentiation

**Review by BNAndras — APPROVED** [review]

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:17]: Does Exercism's website render `math` blocks?

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:29]: Square isn't part of the tests. Should it be here? If yes, should it have a header?

**Review by IsaacG — APPROVED** [review]

File: exercises/complex-numbers/description.md
Comment (ErikSchierboom) [reply exercises/complex-numbers/description.md:17]: I don't think we do

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:29]: Thank you for pointing this out! Since the square of the absolute value isn't part of the tests or required to solve the exercise, I think it might be better to leave it out altogether. Including it could potentially confuse learners by making them think it's something they need to implement.

On a related note, I noticed that the reciprocal isn't tested either. I included it because it was in the previous version of the file, but if you feel it’s unnecessary, I can remove that as well.

Let me know your thoughts, and I’ll make the changes accordingly. 😊

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:17]: @IsaacG, yes, @ErikSchierboom is right! Exercism doesn't render math blocks (as far as I know, since I haven't seen them anywhere), and they don't work in the forum. However, the tests and checks on GitHub require me to specify the block type. I initially thought it would make sense to specify it as a math block, but my original intention was to keep it as plaintext. I didn't specify plaintext initially, though.

Now, I'm wondering whether I should change the block type to plaintext or leave it as is (since Exercism doesn't support math blocks, so I assume it will render it as plain text instead of throwing an error). The reasoning is that if Exercism ever starts supporting math blocks in the future, it might be useful to leave it as is so it renders appropriately.

What do you think?

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:17]: It might be better to change it to 'plaintext' or 'text'. When I preview the file in vscode, I'm getting rendering errors due to the `math` block.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:29]: The square of a complex number could be used in a division operation (the square of the second number is the denominator in both fractions), so it might be preferable to leave it for those who wish to explore that approach:

```(a + i * b) / (c + i * d) = (a * c + b * d)/(c^2 + d^2) + (b * c - a * d)/(c^2 + d^2) * i```

The reciprocal section doesn't seem to be useful at all and can likely be safely removed.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:29]: You're right that the reciprocal doesn't have a direct use in the given context. It might be the case that the author included it initially as a way to offer an alternative approach or to keep the option open for students who want to explore multiplying by the reciprocal, as it essentially performs division. But in terms of the immediate solution, it does seem unnecessary.

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:17]: Thanks, @tasxatzial for the clarification! I agree with your reasoning that switching to plain text seems like the better option, especially given Exercism’s current limitations with not rendering math blocks. Aesthetically, it makes sense to avoid rendering errors and ensure everything displays cleanly. You’re right that even if math blocks are supported in the future, they may not necessarily be labelled as such. Changing it to plaintext would likely be the safest choice for now. I’ll push the changes ASAP.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:29]: True, dividing by a number is the same as multiplying by its reciprocal, just like with real numbers. However, this might not be immediately obvious, especially when working with complex numbers. Looking through community solutions, I noticed that some people use the square method, while none seem to use the reciprocal.

Also, I noticed that in the new instructions, the reciprocal is introduced after the division. This might cause people to focus more on the division formula and ignore the reciprocal as an option. In the original instructions, the reciprocal was mentioned right after multiplication, though it didn't seem to make any difference overall. If we still want people to consider the reciprocal route, we might want to explicitly indicate its connection with the division operation, like so:

`z₁ / z₂ = z₁ * (1 / z₂) = <formula goes here>`

I don't have a strong opinion on this, but since I'm biased due to my background, I'll stay out of the decision. Some might argue that the instructions are too technical and prefer removing both the square and the reciprocal altogether.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:29]: I agree that removing the reciprocal concept could simplify things and make the instructions more straightforward, especially for those who might find it confusing when dealing with complex numbers. Simplifying it would likely make the learning process smoother, so I think it could be a great move. Thanks for sharing your thoughts!

**Comment by jagdish-15** [comment]
I believe this addresses both of the issues mentioned above. @tasxatzial, @ErikSchierboom, @IsaacG, @SleeplessByte, @Cool-Katt, and @BNAndras—please let me know if there’s anything else that needs attention.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:29]: Keep in mind that there are already 5 approvals, so it might be better to leave the discussion on whether to remove the square and reciprocal for another thread. As it stands, I would still suggest moving the reciprocal before the division (as it was in the original instructions) and just adding the `z₁ * (1 / z₂)` to give everyone the chance to use the reciprocal if they want.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:31]: Sure, I've changed it accordingly!

Comment (IsaacG) [reply exercises/complex-numbers/description.md:31]: 1. There's a typo in the header.
2. I think the Square section ought to be dropped. It's not relevant to this exercise. Reciprocal is useful for division, so it can be useful and bumped to a later discussion. I'm not sure why Square would be kept.

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:11]: What makes the absolute value a property and not an operation?

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:17]: This repo uses `text` (117 times) and not `plaintext` (4 times).

Comment (jagdish-15) [reply exercises/complex-numbers/description.md]: Sure, I've changed it to text!

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:29]: Upon reflecting further, I realized that the header isn't necessary at all, as the property in question is the square of the absolute value of the complex number, not the complex number itself. Regarding its significance, this property is widely used in various operations, from division to finding the reciprocal of a complex number. If you examine the formulas, you'll notice it appears in the denominators of both the real and imaginary components.

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:29]: That's correct, all denominators are the square of either the first or the complex number.

Here's a solution that defines it on line 7 and uses it for division on line 29

https://exercism.org/tracks/clojure/exercises/complex-numbers/solutions/fergalbyrne

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:11]: I believe the concept of "key property" isn't very well defined and can be interpreted in different ways. Technically, neither the conjugate nor the absolute value is regarded as a property of complex numbers. Properties refer to characteristics of the numbers, like the associative rule `(a + b) + c = a + (b + c)` for real numbers.

Personally, I'm not too picky about this, given that we're not really teaching math here, and what we already have should be sufficient for this exercise. However, if we want to be accurate, both the conjugate and the absolute value (or square) are simply operations.

Comment (IsaacG) [reply exercises/complex-numbers/description.md:11]: It sounds like you're defining "property" as a function which operates on one complex number and defining "operation" as a function which operates on two complex numbers. In that case, wouldn't "exponent" be a property? Or would it make more sense to not apply this distinction and just call them all "operations"?

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:26]: Should these use superscript 2 and not the `^` symbol, in like with the other codeblocks?

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:26]: The exponentiation still uses the `^` symbol. It might be better to avoid superscripts and subscripts altogether, as they can be a bit too small for some people to read.

Comment (kotp) [reply exercises/complex-numbers/description.md:26]: Consistency over "too small to read" because I can read them with a reader, or make the font size bigger, but being inconsistent makes things "jitter" a bit in what has to be understood.

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:26]: Completely agree. Removing the superscripts and subscripts means that we revert back to how the instructions currently are. So a^2, b^2, z1, z2, e^a etc.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:26]: Sure, I am happy to change it to superscript!

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:11]: The key point I want to highlight is that operations like conjugate, absolute value, exponentiation, and reciprocal are unary operators, while addition, subtraction, multiplication, and division are binary operators for complex numbers. This raises an important question: should unary and binary operators be categorized separately, or should they all be grouped together under a unified "Operations" section?  

My reasoning for placing **absolute value** and **conjugate** in the *Key Properties* section, separate from other operations, is that these two play a foundational role in complex number arithmetic. For example:  
- **Absolute value** is central to defining magnitude, normalization, and division.  
- **Conjugate** is essential for computing reciprocals and simplifying division.  

I acknowledge, however, that both involve computation and are technically unary operations, not inherent "properties" in the strict mathematical sense. As @tasxatzial rightly pointed out, properties typically refer to inherent characteristics like associativity or commutativity.  

That said, my intention in categorizing them as "key properties" was to emphasize their foundational role in complex number operations. If unifying all unary and binary operators under "Operations" better suits the context of this repository, I’m happy to align with that approach.  

Looking forward to your thoughts on whether separating these foundational elements adds clarity or if a unified categorization would be more consistent.

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:26]: Superscripts and subscripts aren't easy for everyone to read. Some people may have low-quality monitors, while others may have impaired eyesight. They pose accessibility challenges, as they are small compared to normal text and disrupt visual alignment, making it harder to follow the content. This creates a similar 'jitter' effect that Kotp mentioned. If Exercism were to implement proper math blocks in the future, subscript and superscript notation would be a great option. For now, they are just a workaround due to the lack of proper math rendering.

Please revert everything to how it was before, as it wasn't an issue previously.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:26]: @tasxatzial I haven’t made any changes yet; I’m still waiting for a conclusion to be drawn!

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:26]: There are two options. Either switch to how it was before, or use superscripts/subscripts. Both of them have no consistency issues. And since there are issues with superscripts/subscripts, it's probably best if we avoid them altogether.

Also, take a look at the complex conjugate `z̅ = a - b * i`. What is the first symbol? A `z`? The line above z is almost touching the letter, making it illegible. The current instructions don't even use a special notation for the conjugate.

All those issues wouldn't exist if Exercism properly supported math.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:26]: We can also represent the conjugate as `z*` if the bar notation causes issues, or we can avoid denoting it altogether. However, I believe this representation should work fine.  

I’m also happy to revert to the original notation and avoid using subscript or superscript if no one has any concerns with that.

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:26]: > However, I believe the `z*` representation should be fine.

This now looks like the multiplication symbol. I'd just assign the conjugate to a variable and use it:

The conjugate `zc` of the complex number `z = a + b * i` is given by...

File: exercises/complex-numbers/description.md
Comment (tasxatzial) [reply exercises/complex-numbers/description.md:11]: Since the exercise requires implementing 'operations', which include both the conjugate and the absolute value, it might be better to move them to the operations section for consistency.

File: exercises/complex-numbers/description.md
Comment (IsaacG) [reply exercises/complex-numbers/description.md:26]: It sounds like people are leaning towards using the `^` symbol. I don't have a strong opinion here. Using `^` sounds like the most consistent option.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:26]: Sure, we’ll use the `^` symbol and avoid using notation for the conjugate altogether to prevent issues with bar rendering and the potential confusion of `z*` with multiplication.

File: exercises/complex-numbers/description.md
Comment (IsaacG) [inline exercises/complex-numbers/description.md:56]: I think that intermediate step is helpful to show. It would be nice to include it in the addition and subtraction formula, too!

**Review by IsaacG — APPROVED** [review]

File: exercises/complex-numbers/description.md
Comment (jagdish-15) [reply exercises/complex-numbers/description.md:56]: I've made these changes, please check them out!

Comment (tasxatzial) [reply exercises/complex-numbers/description.md:56]: I feel that the equations are becoming a bit too long as one-liners. Perhaps consider breaking the following into multiple lines?

```text
z1 + z2 = (a + b * i) + (c + d * i)
        = (a + c) + (b + d) * i

z1 * z2 = (a + b * i) * (c + d * i)
        = (a * c - b * d) + (b * c + a * d) * i

1 / z = 1 / (a + b * i)
      = a / (a^2 + b^2) - b / (a^2 + b^2) * i

z1 / z2 = z1 * (1 / z2)
        = (a + b * i) / (c + d * i)
        = (a * c + b * d) / (c^2 + d^2) + (b * c - a * d) / (c^2 + d^2) * i

e^(a + b * i) = e^a * e^(b * i)
              = e^a * (cos(b) + i * sin(b))
```

The absolute value does not need to be broken as it's already short and readable.

Comment (jagdish-15) [reply exercises/complex-numbers/description.md:56]: Sure

**Comment by jagdish-15** [comment]
Shouldn't we mark this issue as resolved since we have verified that the tests for the Yacht exercise are already in sync, and updates for the other exercises are currently in progress?

**Comment by jagdish-15** [comment]
With this commit, I believe the formatting issues are resolved, and this should be ready to merge. Please let me know if I’ve missed anything.

A big thank you to @ErikSchierboom, @kotp, @Cool-Katt, @BNAndras, and @SleeplessByte, and especially to @tasxatzial and @IsaacG for reviewing this and providing your valuable feedback!

**Review by IsaacG — APPROVED** [review]
Looks good to me!

**Review by kotp — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
